### PR TITLE
NAS-102190 / 11.3 / feat(middlewared/pool): translate a few zfs error codes and document it

### PR DESCRIPTION
**Pull request by william-gr**
Ticket:	NAS-102190

### CODE REVIEW

**Comment by william-gr** [comment]
@themylogin i am not too attached to this solution but seemed a fair compromised into translating a few error codes and have them showing up ok without too much work. Let me know what you think please

File: src/middlewared/middlewared/plugins/zfs.py [hunk 1]
```diff
@@ -64,6 +65,14 @@ def find_vdev(pool, vname):
         children += list(child.children)
 
 
+def libzfs_errcode_mapping(e):
+    return {
+        ZFSError.NOSPC: errno.ENOSPC,
+        ZFSError.NOREPLICAS: errno.ENOTRECOVERABLE,
+        ZFSError.NODEVICE: errno.ENOENT,
+    }.get(e.code) or e.code
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/zfs.py]: I don't like that we are mixing `errno` and `zfs_error` (which AFAIK do not match `errno`s), can it be `errno.EFAULT` by default instead?

Comment (william-gr) [reply src/middlewared/middlewared/plugins/zfs.py]: Its either that or have all errors specified in middleware (or they shown as EUNKNOWN) which I think its worse.

Defaulting to e.code at least let us identify what the original error code was and fix it up.

Thoughts?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/zfs.py]: Can we make it `300 + e.code` for example? And unparse it back in `ErrnoMixin`

Comment (william-gr) [reply src/middlewared/middlewared/plugins/zfs.py]: I dont understand how that helps? e.code for ZFS are already in the 2000+ range.

Comment (themylogin) [reply src/middlewared/middlewared/plugins/zfs.py]: I want to isolate them from errno codes so they don't intersect. If they are already that way, fine.
But `ErrnoMixin` part seems useful to me nevertheless.

Comment (william-gr) [reply src/middlewared/middlewared/plugins/zfs.py]: Still not sure what you mean, are you talking about doing the inverse of `ZFSError.NODEVICE: errno.ENOENT` ?

And if so, why?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/zfs.py]: I want to do this in `ErrnoMixin`:
```

    @classmethod
    def _get_errname(cls, code):
        if 2000 <= code <= 2100:
            return 'EZFS' + ZFSError(code).name    
```

Comment (william-gr) [reply src/middlewared/middlewared/plugins/zfs.py]: Ok, thats much smarter ;)

**Pull request merged**